dcc-repository-song/src/main/java/org/icgc/dcc/repository/song/core/SongProcessor.java

```diff
@@ -337,17 +338,12 @@ Donor getDonor(SongAnalysis a) {
       .setSubmittedSampleId(singletonList(sample.get(sampleSubmitterId)));
   }
 
-  SongFile getSongIndexFile(List<SongFile> files, String name) {
+  Optional<SongFile> getSongIndexFile(List<SongFile> files, String name) {
     val songIndex = files.stream().
       filter(f -> f.get(fileName).equalsIgnoreCase(name))
       .findFirst().orElse(null);
 
-    if (songIndex == null) {
-      return null;
-    }
-
-    return songIndex;
-
+    return Optional.ofNullable(songIndex);
```

Comment:
there is redundancy here. The purpose of `getSongIndexFile` (based on the signature) is to return an `Optional<SongFile>` object. The method starts out by streaming SongFiles, and then outputs an `Optional<SongFile>` object, then the method extracts the value from the optional, and then puts it back into an Optional.

This can be made cleaner by doing the following
```java
Optional<SongFile> getSongIndexFile(List<SongFile> files, String name){
  return files.stream()
    .filter(f -> f.get(fileName).equalsIgnoreCase(name))
    .findFirst();
}
```
In addition, it is good practice to make stateless methods static. It makes it immediately clear to the developer after you, that the method does not depend on any state, so this would mean making the `getSongIndexFile` method `private static`